### PR TITLE
Add some HISTORY sections to documentation for newer 3.0 related files.

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
The intention here would be to update these sections when the files are modified.

File: doc/man7/OSSL_PROVIDER-default.pod [hunk 1]
```diff
@@ -252,6 +252,7 @@ L<OSSL_PROVIDER-base(7)>
 =head1 HISTORY
 
 The RIPEMD160 digest was added to the default provider in OpenSSL 3.0.7.
+All other functionality was added to OpenSSL 3.0.
```
Comment (paulidale) [inline doc/man7/OSSL_PROVIDER-default.pod:255]: Blank line above is typical.

File: doc/man5/fips_config.pod [hunk 2]
```diff
@@ -107,6 +107,10 @@ See L<config(5)>.
 L<config(5)>
 L<openssl-fipsinstall(1)>
 
+=head1 HISTORY
+
+This functionality was added to OpenSSL 3.0.
```
Comment (paulidale) [inline doc/man5/fips_config.pod:112]: We typically write _added in_ rather than _added to_.
There are exceptions but they are, by far, the minority.

**Review by paulidale — CHANGES_REQUESTED** [review]

File: doc/man5/fips_config.pod (hunk 2 above)
Comment (slontis) [reply doc/man5/fips_config.pod]: I block copied it from another file.. Updated them all.

**Review by paulidale — APPROVED** [review]

File: doc/man7/EVP_KEM-EC.pod [hunk 3]
```diff
@@ -66,6 +66,10 @@ L<EVP_KEYMGMT(3)>,
 L<EVP_PKEY(3)>,
 L<provider-keymgmt(7)>
 
+=head1 HISTORY
+
+This functionality was added in OpenSSL 3.2.
+
```
Comment (t8m) [inline doc/man7/EVP_KEM-EC.pod:72]: It would be nice to split this out into a separate commit so we might be able to cherry-pick to 3.1/3.0?

File: doc/man7/fips_module.pod [hunk 4]
```diff
@@ -460,6 +460,11 @@ L<OSSL_PROVIDER_get0_name(3)>.
 
 L<migration_guide(7)>, L<crypto(7)>, L<fips_config(5)>
 
+=head1 HISTORY
+
+The fips module guide was created for use with the new FIP provider
```
Comment (t8m) [inline doc/man7/fips_module.pod:465]: Typo: FIPS

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man7/EVP_KEM-EC.pod (hunk 3 above)
Comment (slontis) [reply doc/man7/EVP_KEM-EC.pod]: Done- separate PR created,

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master, 3.0, and 3.1 branches. There was a mistake that I've noticed only when cherry-picking to 3.1 branch corrected by #19725
